File: BKJ/듣보잡/pable91.py
# 듣보잡
# 1. map 은 반복가능한 객체에 함수(lambda, int)를 활용가능
# 2. 반복 불가능한 타입(int 정수)을 순회하려면 range 키워드를 써야함
# 3. {}는 set, []는 리스트, ()는 튜플
# 4. & 는 두 집합간의 교집합을 구할 때 사용


a, b = map(int, input().split())

arr1 = {input() for _ in range(a)}
arr2 = {input() for _ in range(b)}

result = sorted(arr1 & arr2)

print(len(result))

for elem in result:
  print(elem)

File: BKJ/소가 길을 건너간 이유 1/pable91.kt
/**
 * 소가 길을 건너간 이유 1
 * https://www.acmicpc.net/problem/14467
 *
 * 1. 입력받은 정수만큼 for문으로 입력
 * 2. 소들의 위치를 기억할 배열 선언
 * 3. 소돌의 위치 이동 횟수를 기억할 배열 선언
 * 4. 이전 위치와 다르면 위치 이동 횟수 배열에서 +1 한다
 * 5. 소들의 위치 이동 횟수 배열을 모두 sum 한다
 *
 */

fun main() {

    val input = readln().toInt()
    val arr = IntArray(100) {-1}
    val cowMoveCount = IntArray(100) {0}
    for (i in 0 until input) {
        val pairs = readLine()!!.split(" ")
            .map { it.toInt() }

        if(arr[pairs.first()] == -1) {
            cowMoveCount[pairs.first()] = 0
        }
        else if (arr[pairs.first()] != pairs.last()) {
            cowMoveCount[pairs.first()] += 1
        }
        arr[pairs.first()] = pairs.last()
    }

    var result = 0;
    for (elem in cowMoveCount) {
        result += elem;
    }
    print(result);
}